src/app/core/services/grouping/group.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { Group } from '../../models/github/group.interface';
+import { GroupingContextService } from './grouping-context.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GroupService {
+  /** Users to show as columns */
+  public groups: Group[] = [];
+
+  /** The list of users with 0 issues (hidden) */
+  hiddenGroups: Group[] = [];
+
+  constructor(public groupingContextService: GroupingContextService) {}
+
```

Comment:
Since the GroupingContextService is only used for logic related to groups, why not remove it entirely from the issue viewer component? The only balance is the initialization of GroupingContextService and the initialization of available groups. We can move this logic to GroupService as well.

Reply:
No, I mean we can move the usage of groupingContextService from issueViewer to groupService, since the groupingContextService is only related to groups. You just need to move the initialization of GroupingContextService and available groups from issueViewer to groupService.

Reply:
Understood, but what about the other lines in `issues-viewer.component` which still use `groupingContextService`? For example:

```
    this.groupByChangeSubscription = this.groupingContextService.currGroupBy$.subscribe((newGroupBy) => {
      this.initialize();
    });
```

Reply:
@kokerinks Fine, I think is quite hard to entirely move it. Then, just leave it as it is for now. Just let me know after you resolve the conflicts. Thanks!